application-collabora-ui/src/main/resources/Collabora/Code/UI.xml

```diff
@@ -539,20 +539,37 @@ require(['jquery', 'xwiki-l10n!collabora-attachment'],function($, l10n) {
     </property>
     <property>
       <content>{{velocity output="false"}}
+#macro (getPageEditURL $pageEditURL)
+  #set ($editWiki = $services.wiki.currentWikiId)
+  #if ($services.collabora.getConfiguration().editUsingMainWiki())
+    #set ($editWiki = $services.wiki.mainWikiId)
+  #end
+  #set ($pageEditRef = $services.model.createDocumentReference($editWiki, ['Collabora', 'Code'], 'Main'))
+  ## In case the main wiki was selected as edit endpoint, but the application is not installed on it, fallback on the
+  ## current wiki as endpoint.
+  #if (!$xwiki.exists($pageEditRef))
+    #set ($pageEditRef = $services.model.createDocumentReference($currentWikiId, ['Collabora', 'Code'], 'Main'))
+  #end
+  #set ($pageEditURL = $xwiki.getDocument($pageEditRef).getURL('get'))
+#end
 #macro (tableButtonTemplate)
   #set ($logoURL = $xwiki.getDocument('Collabora.Code.UI').getAttachmentURL('collabora-symbol.svg'))
   #set ($label = $escapetool.xml($services.localization.render("collabora.attachment.title")))
+  #getPageEditURL($pageEditURL)
   ## After the application starts depending on a version of XWiki &gt;= 14.6, this code should be moved inside an
   ## org.xwiki.platform.attachment.actions UIX.
-  &lt;a class="action collaboraEdit" href=""&gt;
+  &lt;a class="action collaboraEdit" href="" data-page-url="$pageEditURL"&gt;
```

Comment:
``data-page-url`` is not very semantic. "What page?" I'm more in favour of something like ``data-action-url`` or ``data-edit-url`` or ``data-service-url``.